### ISSUE TITLE
Add multiple region per AreaShop support

### ISSUE THREAD

**Issue by FireLordZuko** [NLthijs48/AreaShop#131]
Hey,

First of all, great plugin! It's clear that you are putting a lot of effort into this. I have a request for a new feature, and would like to see this implemented as soon as possible as it is so incredibly convenient for the customization within your plugin.

We have houses with various sizes and shapes, houses that cannot fit inside of 1 single WorldEdit selection, but require multiple worldedit selection. As far as I know, there is no way in WorldGuard to add multiple WorldEdit selections to 1 single WorldGuard region. Would it be possible to at least add this support to AreaShop.

An example of an implementation would be that we could for example have 5 WorldGuard region and put them as a child of a parent WorldGuard region. We could then insert the name of the parent WorldGuard region as an AreaShop and we could configure it to be recognized as a parent instead of a single WorldGuard region. AreaShop would then look for the children instead of the parent and would form a shop out of all these regions combined. This is an example, you could obviously tackle it however you'd like to!

Thanks in advance!

**Comment by Fuechschen** [comment]
If i understand you right, you want to combine multiple regions into one?

You could simply use a ploygonal selector in WorldEdit to produce region of any shape and size. Of course this only works if the regions are connected.

**Comment by FireLordZuko** [comment]
Not exactly, the buildings we have (together with the roofs etc), are a little oddly shaped when it comes to selecting them as a polygon.

**Comment by NLthijs48** [comment]
There is a workaround available for this problem, however I'm interested at solving this correctly at some point. See #53 and [this comment](http://dev.bukkit.org/bukkit-plugins/regionbuyandrent/?comment=614) for the work around.

The work around uses clever naming to assign extra regions, this might be an alternative approach to this problem. Setting the parent of all regions used to make a combined region is relatively simple, but if only using a clever naming scheme is possible to do the same thing then this might be preferred. I'm not sure about a concrete solution yet though, maybe something like this in `default.yml`:
```yaml
general:
  extraRegions:
  - '%region%-right'
  - '%region%-left'
```
Which would mean that if you create the regions `shop`, `shop-left` and `shop-right`, and then add the `shop` region to AreaShop, then it will automatically sync the other regions with the first one (basically means applying the region flags and handling schematic save/restore). This approach means setting it up one time, and then simply creating regions with the correct names to get it going.

What would you think about such a solution?

**Comment by FireLordZuko** [comment]
It'd be a little more work but it works for me :)

**Comment by NLthijs48** [comment]
Do you mean more work because you need to add these lines to the config file or because you would have to change the names of the regions?

**Comment by FireLordZuko** [comment]
My shops are not simply added by right or left conventions, so I'd like to use my own personal naming scheme for every individual shop if possible. Having everything under 1 parent ensures you that you can easily just change the regions which are the children underneath this parent and makes it much more customizable.

The solution you provided in the comment above is fine however, the workaround which is currently available is a hassle to work with if you have around 20 regions per areashop for example.

**Comment by NLthijs48** [comment]
Of course you could use different names in the config, but your point that managing ~10 different region per shop is hard is true. So that would indeed require a different solution, maybe we could combine the two and simply see a region as connected if it is a child of the actual main AreaShop region (as you suggest) and if it is listed in the config. The reason to keep my approach is if you have a lot of region to make, which all have a couple of connected regions, then it would save you the commands to set it as child.

Adding both would not require much extra work, since introducing the concept of connected regions is the main change that is required, how those regions are added/determined is not much work to implement.

Thanks for your input.

**Comment by FireLordZuko** [comment]
Thank you very much for being willing to implement this, any idea when you think this could be ready?

**Comment by NLthijs48** [comment]
I'm sorry but I cannot give any estimate for a completion time, I'm quite busy with my study on the university. Currently working on the message system (#18) in AreaShop, so that for sure needs to be finished first.
